src/types.ts

```diff
@@ -1939,14 +1939,12 @@ export type ValidationTargets<T extends FormValue = ParsedFormValue, P extends s
 //////                            //////
 ////////////////////////////////////////
 
-type ParamKeyName<NameWithPattern> = NameWithPattern extends `${infer Name}{${infer Rest}`
-  ? Rest extends `${infer _Pattern}?`
-    ? `${Name}?`
-    : Name
-  : NameWithPattern
-
 type ParamKey<Component> = Component extends `:${infer NameWithPattern}`
-  ? ParamKeyName<NameWithPattern>
+  ? NameWithPattern extends `${infer Name}{${infer Rest}`
+    ? Rest extends `${infer _Pattern}?`
+      ? `${Name}?`
+      : Name
+    : NameWithPattern
```

Comment:
Changing to this will result in a slight performance increase.

```suggestion
  ? NameWithPattern extends `${infer Name}{${string}}${infer Rest}`
    ? `${Name}${Rest extends `?` ? '?' : ''}`
    : NameWithPattern
  : never;
```

Reply:
```llvm
runtime: bun 1.1.34 (x64-win32)

benchmark                   avg (min … max) p75 / p99    (min … top 1%)
------------------------------------------- -------------------------------
Before                       566.16 µs/iter 562.20 µs  █
                      (301.30 µs … 4.42 ms)   2.33 ms  █▃
                    (  0.00  b …   2.39 gb)  76.41 mb ███▅▄▃▂▂▁▁▁▂▁▁▁▁▁▁▁▁▁
After                        334.73 µs/iter 320.10 µs  █
                      (254.40 µs … 4.39 ms) 967.10 µs  █
                    (  0.00  b …   1.82 gb)  23.02 mb ▆█▇▃▃▂▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁
After2                       311.22 µs/iter 300.60 µs  █
                      (245.10 µs … 3.10 ms) 639.50 µs  ██
                    (  0.00  b … 516.00 mb)  20.57 mb ▆██▆▃▃▃▃▂▁▁▁▁▁▁▁▁▁▁▁▁
```